lib/console/ai/evidence/component/stateful_set.ex

```diff
@@ -1,9 +1,68 @@
 defmodule Console.AI.Evidence.Component.StatefulSet do
   use Console.AI.Evidence.Base
+  alias Console.AI.Evidence.Component.PersistentVolumeClaim
 
-  def hydrate(%AppsV1.StatefulSet{metadata: %{namespace: ns}, spec: %{selector: selector}}) do
-    list_pods(ns, selector)
-    |> default_empty(&pod_messages("statefulset", &1))
+  def hydrate(%AppsV1.StatefulSet{metadata: %{namespace: ns}, spec: %{selector: selector}} = sts) do
+    with {:ok, pod_evidence} <- list_pods(ns, selector) |> default_empty(&pod_messages("statefulset", &1)),
+         pvc_evidence <- pvc_messages(sts) do
+      evidence(pod_evidence ++ pvc_evidence)
+    else
+      _ -> {:ok, []}
+    end
   end
+
   def hydrate(_), do: {:ok, []}
+
+  defp pvc_messages(%AppsV1.StatefulSet{
+         metadata: %{name: sts_name, namespace: ns},
+         spec: %{replicas: replicas, volume_claim_templates: vcts}
+       })
+       when is_list(vcts) do
+    vcts
```

Comment:
this should probably be written as a reduce into the Context struct, eg:

```elixir
Enum.reduce(vcts, Context.new(history), fn vct, acc ->
  Enum.reduce(0..(replicas - 1), acc, fn i, acc ->
     case fetch_pvc_evidence("#{vct.metadata.name}-#{sts_name}-#{i}") do
       {:ok, ctx} -> Context.merge(acc, ctx)
       _ -> acc
     end
  end)
end)
```

(I don't believe the Context.merge function has been implemented yet though, and needs to handle both the case where its given a context struct or just a message list)